src/Dynode/model/core/node.ts

```diff
@@ -64,25 +66,35 @@ class Node extends VEventTarget {
   /**
    * A collection of OutputSocket objects.
    */
-  outputs = new SocketCollection<OutputSocket>();
+  outputs = new SocketCollection<OutputSocket<unknown>>();
+
+  /**
+   * A helper for retrieving an optional SocketValueType.
+   *
+   * The main purpose is to provide an argument for socket which can take undefined and
+   * assume the default or a name can be passed which then is extracted from engine. This
+   * done so it is possible to add default type sockets without an engine.
+   *
+   * @param name ValueSocketType name that is stored in engine
+   */
+  private getValueType(name?: string): SocketValueType<unknown> | undefined {
```

Comment:
getSocketValueType()

Reply:
But I think it is a bit awkward that Node class needs this helper function. Since the types live in the engine, it would be more natural to do something like this.engine.getSocketValueType(), instead of introducing this helper function. One of the reasons is that node is not aware of the network it is part of, and access to the engine should be through the associated network. And fixing that would indeed be something that has to be done in a separate PR.